src/js/index.js

```diff
@@ -19,8 +22,45 @@ reachData().then((data) => data.forEach(
                <P class="name">${e.show.name}</P>
                <i class="far fa-heart"></i>
            </div>
-           <button>Comments</button>
+           <button class="comments" id="${e.show.id}">Comments</button>
            </div>
             `;
   },
 ));
+
+const getDataFromApi = (id) => {
+  reachData().then((data) => {
+    data.forEach((el) => {
+      if (el.show.id === id) {
+        modalContent.innerHTML = `
+           <div class="card">
+           <img src="${el.show.image.medium}" alt="">
+           <span>Summary:</span>
+           <p>${el.show.summary}</p>
+           <span>language:</span>
+           <p>${el.show.language}</p>
+
+           <div class="name-icon">
+               <p class="name">${el.show.name}</p>
+               <i class="far fa-heart"></i>
+           </div>
+           <button class="comments" id="${el.show.id}">Comments</button>
+           </div>
+            `;
+      }
+    });
+  });
+};
+
+body.addEventListener('click', (e) => {
+  if (e.target.className === 'comments') {
+    modal.style.display = 'block';
+    getDataFromApi(e.target.id);
+  }
+  if (e.target.className === 'close') {
+    modal.style.display = 'none';
+  }
+  if (e.target === modal) {
+    modal.style.display = 'none';
+  }
+});
```

Comment:
- [ OPTIONAL] I would recommend you to use the Switch  case js format to replace many if conditions in your ``addEventListner method,  it will improve you codes readablity. 
![switch](https://user-images.githubusercontent.com/43877689/161769286-29401515-8176-47a6-a78f-310ff793eebd.PNG)
For more information about [switch case](https://www.w3schools.com/js/js_switch.asp)